internal/apis/certmanager/validation/certificate.go

```diff
@@ -224,3 +214,26 @@ func ValidateDuration(crt *internalcmapi.CertificateSpec, fldPath *field.Path) f
 	}
 	return el
 }
+
+func validateAdditionalOutputFormats(crt *internalcmapi.CertificateSpec, fldPath *field.Path) field.ErrorList {
+	var el field.ErrorList
+
+	if !utilfeature.DefaultFeatureGate.Enabled(feature.AdditionalCertificateOutputFormats) {
+		if len(crt.AdditionalOutputFormats) > 0 {
+			el = append(el, field.Forbidden(fldPath.Child("additionalOutputFormats"), "feature gate AdditionalCertificateOutputFormats must be enabled"))
+		}
+		return el
+	}
+
+	// Ensure the set of output formats is unique, keyed on "Type".
+	aofSet := sets.NewString()
+	for _, val := range crt.AdditionalOutputFormats {
```

Comment:
May just be me, but does this read easier if we switch it around to be:


```
func validateAdditionalOutputFormats(crt *internalcmapi.CertificateSpec, fldPath *field.Path) field.ErrorList {
	if utilfeature.DefaultFeatureGate.Enabled(feature.AdditionalCertificateOutputFormats) {
 		var el field.ErrorList
 		// .. perform new validation code .. //
 		return el
 	} else if len(crt.AdditionalOutputFormats) > 0 {
 		// .. return forbidden .. //
                return el
        }
}
```

It took me a little while to realise that we'd *not* validate the `Type` field if the feature gate is disabled. Perhaps one instance where an `else` in Go _is_ useful? (or maybe I just need to read more carefully 😄

Reply:
This was actually intentional. I wanted to have the feature gate check to be the "first" thing in the function- both so it is clear that check doesn't get lost or mangled in other logic, and when/if the feature is graduated we only need to delete that one code block above.